core/templates/dev/head/domain/exploration/StateObjectFactory.js

```diff
@@ -19,36 +19,47 @@
 
 oppia.factory('StateObjectFactory', [
   'AnswerGroupObjectFactory', 'InteractionObjectFactory',
-  function(AnswerGroupObjectFactory, InteractionObjectFactory) {
-  var State = function(name, classifierModelId, contentBackendList,
-    interactionBackendDict, paramChangesBackendList) {
-    this.name = name;
-    this.classifierModelId = classifierModelId;
-    this.content = contentBackendList;
-    this.interaction = InteractionObjectFactory.create(interactionBackendDict);
-    this.paramChanges = paramChangesBackendList;
-  };
+  'ContentObjectFactory',
+  function(AnswerGroupObjectFactory, InteractionObjectFactory,
+    ContentObjectFactory) {
+    var State = function(name, classifierModelId, contentBackendList,
```

Comment:
Maybe that should wait until after there are outcome and rule objects? Otherwise I'd be passing in objects to the constructor via createFromBackendDict and dicts to the constructor via createNew, so I'd need two separate constructors. But I could do it for StateObjectFactory